.eslintrc.js

```diff
@@ -1,17 +1,11 @@
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
-const PACKAGE_NAME = require('./package').name;
-const SYMLINK_LOCATION = path.join(__dirname, 'node_modules', PACKAGE_NAME);
-
-// Symlink node_modules/{package name} to this directory so that ESLint resolves this plugin name correctly.
-if (!fs.existsSync(SYMLINK_LOCATION)) {
-  fs.symlinkSync(__dirname, SYMLINK_LOCATION);
-}
+// Register ourselves as a plugin to avoid `node_modules` trickery.
+const Plugins = require('eslint/lib/config/plugins');
+Plugins.define('prettier', require('.'));
```

Comment:
I think this approach is going to break after https://github.com/eslint/eslint/pull/8465 is merged, which is likely to happen in ESLint 4. (The change makes `Plugins` an ES6 class rather than a global mutable object, to prevent unrelated modules from conflicting with each other if they both happen to be using eslint in the same process.)